maestro-utils/src/test/kotlin/SocketUtilsTest.kt

```diff
@@ -0,0 +1,75 @@
+import maestro.utils.SocketUtils
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertFalse
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.Assertions.assertThrows
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+import java.net.Inet4Address
+import java.net.InetAddress
+import java.net.NetworkInterface
+import java.net.ServerSocket
+
+class SocketUtilsTest {
+
+    @Test
+    fun `nextFreePort should return a free port within the specified range`() {
+        val from = 5000
+        val to = 5100
+        val port = SocketUtils.nextFreePort(from, to)
+
+        assertTrue(port in from..to)
+    }
+
+    @Test
+    fun `nextFreePort should throw IllegalStateException if no ports are available`() {
+        val from = 5000
+        val to = 5001
+
+        val serverSocket1 = ServerSocket(from)
+        val serverSocket2 = ServerSocket(to)
+
+        try {
+            assertThrows(IllegalArgumentException::class.java) {
+                SocketUtils.nextFreePort(from, to)
+            }
+        } finally {
+            serverSocket1.close()
+            serverSocket2.close()
+        }
+    }
+
+    @Test
+    fun `localIp should return a non-loopback IPv4 address`() {
+        val ip = SocketUtils.localIp()
+
+        assertNotNull(ip)
+        assertTrue(ip.startsWith("192") || ip.startsWith("10") || ip.startsWith("172"))
```

Comment:
This assertion fails for me when testing this locally as the IP resolves to `127.0.0.1` instead

Reply:
Resolved here https://github.com/mobile-dev-inc/maestro/pull/2081/commits/08c9a16a848a1821de0c3429e86f71b5fa1355b1